components/ws-manager-bridge/src/bridge.ts

```diff
@@ -219,10 +219,20 @@ export class WorkspaceManagerBridge implements Disposable {
                     break;
                 case WorkspacePhase.STOPPING:
                     instance.status.phase = "stopping";
+                    if (!instance.stoppingTime) {
+                        // The first time a workspace enters stopping we record that as it's stoppingTime time.
+                        // This way we don't take the time a workspace requires to stop into account when
+                        // computing the time a workspace instance was running.
+                        instance.stoppingTime = new Date().toISOString();
+                    }
                     break;
                 case WorkspacePhase.STOPPED:
-                    instance.stoppedTime = new Date().toISOString();
                     instance.status.phase = "stopped";
```

Comment:
`instance.stoppedTime = new Date().toISOString();` went missing here, should be re-added.